### PR TITLE
Recipe App 💪 

### PR DESCRIPTION
**Pull request by leehaney254**
- Used Ruby gems as a software packages system.
- Installed Ruby on Rails framework.
- Created Rails RESTful design and router.
- Used controllers to handle requests and render empty views.
- Worked with Rails naming conventions.
- Worked with params from browser requests in a safe way.
- Used preprocessed HTML file with embedded Ruby code.
- Used layouts and templates for shared content.
- Used database migration files to maintain database schema.
- Used validations for models.
- Secured our app from n+1 problems.
- Wrote SQL queries with ActiveRecord.
- Created associations between models.
- Created user login.
- Implemented authorization and authentication.
- Used Devise gem for authentication.
- Wrote unit tests using Rspec
- Wrote integration tests with capybara gem

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+<table class='w-3/6 mt-5 text-sm text-left text-gray-500 dark:text-gray-400'>
+  <thead class="text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400">
+    <tr>
+      <th scope="col" class="px-6 py-3">Name</th>
+      <th scope="col" class="px-6 py-3">Quantity</th>
+      <th scope="col" class="px-6 py-3">Value</th>
+      <th scope="col" class="px-6 py-3">Actions</th>
+    </tr>
+  </thead>
+  <tbody>
```
Comment (mikethreels) [inline app/views/recipes/show.html.erb:44]: - [ ] when adding a food to a recipe with another quantity than 1 the application should calculate the total price for that food in the recipe, in your application it doesn't as you can see in below printscreen:
image

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Great job making use of partials

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (leehaney254) [reply app/views/recipes/show.html.erb:44]: Thank you very much 💯 . We have fixed it.

**Review by afanimartin — APPROVED** [review]
Hi @leehaney254, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Nice fixing the cart error

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**